### PR TITLE
update checkout action to v3

### PR DESCRIPTION
**Pull request by yizshi**
## Description

Update github action version to avoid node12 deprecation warnign
## Motivation and Context

CI warns about node12 being deprecated 
## How Has This Been Tested?

CI should pass without any warning 
## Types of Changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

### CODE REVIEW

File: .github/workflows/java-ci.yml [hunk 1]
```diff
@@ -17,10 +17,11 @@ jobs:
         java: [8, 11, 17]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
       - name: Setup java
-        uses: actions/setup-java@v1
+        uses: actions/setup-java@v2
         with:
+          distribution: "zulu"
```
Comment (yizshi) [inline .github/workflows/java-ci.yml:24]: https://github.com/actions/setup-java
> V2 requires you to specify distribution along with the version. V1 defaults to Azul Zulu OpenJDK

Just use the default distribution we have been using so far.

Comment (yizshi) [reply .github/workflows/java-ci.yml:24]: side notes, seems setup-java v2 havent get into node update. V2 already the latest, so even we still have warning now it should be gone once setup-java update their node version

File: .github/workflows/java-ci.yml [hunk 2]
```diff
@@ -17,10 +17,11 @@ jobs:
         java: [8, 11, 17]
 
     steps:
-      - uses: actions/checkout@v2
+      - uses: actions/checkout@v3
```
Comment (AaronAtDuo) [inline .github/workflows/java-ci.yml:20]: Looks like they're up to 4 actually https://github.com/marketplace/actions/checkout should we just go ahead and use that?

Comment (AaronAtDuo) [reply .github/workflows/java-ci.yml:20]:
```suggestion
      - uses: actions/checkout@v4
```

**Review by AaronAtDuo — APPROVED** [review]

**Pull request merged**